src/aliceVision/sfm/pipeline/ReconstructionEngine.cpp

```diff
@@ -56,15 +59,29 @@ void retrieveMarkersId(sfmData::SfMData& sfmData)
 
         const auto obs = landmark.observations.begin();
         const feature::Regions& regions = regionPerView.getRegions(obs->first, landmark.descType);
-        const feature::CCTAG_Regions& cctagRegions = dynamic_cast<const feature::CCTAG_Regions&>(regions);
-        const auto& d = cctagRegions.Descriptors()[obs->second.id_feat];
-        for (int i = 0; i < d.size(); ++i)
-        {
-            if (d[i] == 255)
+        const feature::CCTAG_Regions* cctagRegions = dynamic_cast<const feature::CCTAG_Regions*>(&regions);
+        const feature::APRILTAG_Regions* apriltagRegions = dynamic_cast<const feature::APRILTAG_Regions*>(&regions);
```

Comment:
@fabiencastan Not sure what happens here, but i have the impression that the 2 dyn cast can work even if the the input regions are not the correct type.
Maybe we should have a "special" region type "markers" from which all the other (cctag, april...) inherit and refactor the code to retrieve the iID?

Reply:
I changed the existing dynamic cast from reference to pointer. If the input regions do not have the requested type the result will be a null pointer. This is used in the following if statements to execute the piece of code appropriate for the marker type.
I agree that some kind of marker polymorphism might be good once there are multiple types of markers, but I did not dare making any such deep changes together with this otherwise rather shallow (in terms of change of the code base) pull request...